kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaTopicManager.java

```diff
@@ -200,50 +173,20 @@ public InternalServerCnx getInternalServerCnx() {
     // this method do the real lookup into Pulsar broker.
     // retFuture will be completed with null when meet error.
     private void lookupBroker(String topicName,
-                              Backoff backoff,
                               CompletableFuture<InetSocketAddress> retFuture) {
         try {
-            // If channel is closing, complete a null to avoid brings in further handling.
-            rwLock.readLock().lock();
-            try {
-                if (closed) {
-                    retFuture.complete(null);
-                    if (log.isDebugEnabled()) {
-                        log.debug("[{}] Return null for getTopic({}) since channel closing",
-                            requestHandler.ctx.channel(), topicName);
-                    }
-                    return;
-                }
-            } finally {
-                rwLock.readLock().unlock();
-            }
-
             ((PulsarClientImpl) pulsarService.getClient()).getLookup()
-                .getBroker(TopicName.get(topicName))
-                .thenAccept(pair -> {
-                    checkState(pair.getLeft().equals(pair.getRight()));
-                    retFuture.complete(pair.getLeft());
-                })
-                .exceptionally(th -> {
-                    long waitTimeMs = backoff.next();
-
-                    if (backoff.isMandatoryStopMade()) {
-                        log.warn("[{}] getBroker for topic {} failed, retried too many times {}, return null."
-                                        + " throwable: ", requestHandler.ctx.channel(), topicName, waitTimeMs, th);
-                        retFuture.complete(null);
-                    } else {
-                        log.warn("[{}] getBroker for topic failed, will retry in {} ms. throwable: ",
-                            topicName, waitTimeMs, th);
-                        requestHandler.getPulsarService().getExecutor()
-                            .schedule(() -> lookupBroker(topicName, backoff, retFuture),
-                                waitTimeMs,
-                                TimeUnit.MILLISECONDS);
-                    }
-                    return null;
-                });
+                    .getBroker(TopicName.get(topicName))
+                    .thenAccept(pair -> {
+                        checkState(pair.getLeft().equals(pair.getRight()));
+                        retFuture.complete(pair.getLeft());
+                    })
+                    .exceptionally(th -> {
+                        return null;
```

Comment:
`exceptionally` here is meaningless, should you use `retFuture.completeExceptionally(th)` or `retFuture.complete(null)` here? And it's better to add warn logs here. By the way, I think use `whenComplete` is better than `thenAccept` and `exceptionally` here, because we don't care the future returned by `thenAccept`/`exceptionally`.